assignments/lambda-classes.js

```diff
@@ -1 +1,190 @@
 // CODE here for your Lambda Classes
+class Person{
+    constructor(attributes) {
+        this.name = attributes.name;
+        this.location = attributes.location;
+        this.age = attributes.age;
+        this.gender = attributes.gender;
+    }
+    speak() {
+        return `Hello, my name is ${this.name}, I am from ${this.location}`;
+    }
+}
+
+class Instructor extends Person {
+    constructor(personAttributes){
+        super(personAttributes);
+        this.favLanguage = personAttributes.favLanguage;
+        this.specialty = personAttributes.specialty;
+        this.catchPhrase = personAttributes.catchPhrase;
+    }
+    demo(subject) {
+        return `Today we are learning about ${subject}`;
+    }
+    grade(Student, subject) {
+        return `${Student.name} receives a perfect score on ${subject}`;
+    }
+    finalGrade() {
+        return (this.grade + function getRandomInt(min, max) {
+            min = Math.ceil(min);
+            max = Math.floor(max);
+            return Math.floor(Math.random() * (max-min)) + min;
```

Comment:
Alright so main issue with this is that you're trying to access grade, but you're not passing anything in. In this case, we want to pass in the student object so we can have access to the student's grades.

so for instance:
finalGrade(student) {
console.log(student.grade);
// do something
}

and then when you go to invoke it, pass the student whose grade you want to access:
lan.finalGrade(perrin);